frappe/integrations/doctype/paytm_settings/paytm_settings.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import json
+import requests
+from six.moves.urllib.parse import urlencode
+
+import frappe
+from frappe.model.document import Document
+from frappe import _
+from frappe.utils import get_url, call_hook_method, cint, flt, cstr
+from frappe.integrations.utils import create_request_log, create_payment_gateway
+from frappe.utils import get_request_site_address
+from paytmchecksum import generateSignature, verifySignature
+from frappe.utils.password import get_decrypted_password
+
+class PaytmSettings(Document):
+	supported_currencies = ["INR"]
+
+	def validate(self):
+		create_payment_gateway('Paytm')
+		call_hook_method('payment_gateway_enabled', gateway='Paytm')
+
+	def validate_transaction_currency(self, currency):
+		if currency not in self.supported_currencies:
+			frappe.throw(_("Please select another payment method. Stripe does not support transactions in currency '{0}'").format(currency))
```

Comment:
Shouldn't this be Paytm and not Stripe:

```suggestion
			frappe.throw(_("Please select another payment method. Paytm does not support transactions in currency '{0}'").format(currency))
```

Reply:
also `self.supported_currencies`, shouldn't it be just `supported_currencies`?

Reply:
![Screenshot 2020-07-24 at 8 15 21 PM](https://user-images.githubusercontent.com/14824451/88404304-5842c180-cdeb-11ea-8651-45d739645e17.png)